cluster-agent/controllers/argoproj.io/application_controller.go

```diff
@@ -41,16 +48,171 @@ type ApplicationReconciler struct {
 //
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.9.2/pkg/reconcile
+
 func (r *ApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	contextLog := log.FromContext(ctx)
 
-	app := appv1.Application{}
+	// creating databaseQueries object to fetch all the queries functionality
+	// TODO: DEBT - Convert Unsafe methods to default and unchecked
+	databaseQueries, err := database.NewUnsafePostgresDBQueries(true, false)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// retrieve the application CR from the databases and store the value in app object
+	var allApplicationinDb *[]database.Application
+
+	// TODO: DEBT - Convert Unsafe methods to default and unchecked
+	errFetchAllApp := databaseQueries.UnsafeListAllApplications(ctx, allApplicationinDb)
+	if errFetchAllApp != nil {
+		return ctrl.Result{}, errFetchAllApp
+	}
 
+	app := appv1.Application{}
 	if err := r.Get(ctx, req.NamespacedName, &app); err != nil {
 		contextLog.Error(err, "Unable to find Application: "+app.Name)
+
+		// This will check whether the app.Name exists in the fetched lists from Application databases
+		checkAppinDb, dbAppId := contains(*allApplicationinDb, app.Name)
+		if checkAppinDb {
+			// Todo: to decide what will be Application_id from Namespace Application CR perspective
+			// To propose: fetch the list of application from the database and store it in an array, similarly if there is a way to fetch all the application in namespace find that
+			// store that in an array, if the database.name dosen't exists in application namespace create an entry there! (But the follow up question being what if the Application Name is repicated,
+			// should we set Application Name entry in database to be UNIQUE?)
+
+			// Case 1: If nil, then Application CR is present in the database, hence create a new Application CR in the namespace
+			dbApplication := &database.Application{
+				Application_id: dbAppId,
+			}
+			// this proves that the application does exists, create in namespace
+			errGetApp := databaseQueries.UnsafeGetApplicationById(ctx, dbApplication)
+			if errGetApp != nil {
+				return ctrl.Result{}, errGetApp
+			}
+
+			application, errConvert := dbAppToApplicationCR(dbApplication)
+			if errConvert != nil {
+				return ctrl.Result{}, errConvert
+			}
+			errAddToNamespace := r.Create(ctx, application, &client.CreateOptions{})
+			if errAddToNamespace != nil {
+				return ctrl.Result{}, errAddToNamespace
+			}
```

Comment:
Proposal on how to check for the case when the application does not exists in the namespace but in db!

Fetch the list of application from the database and store it in an array!
Then, while taking app.Name as source of identification from the namespace, look in the provided array, the function contains will return a bool value with the Application ID, which will help us to create those application CRs in the namespace with the help of dbAppToApplicationCR method and client create functionality!

This chunk of code still need to be tested properly, but is syntactically correct. Open to suggestions if any.

Note: this is a hack to get this resolved meanwhile, will be researching more for a concrete solution in future!